openmc/mgxs_library.py

```diff
@@ -457,61 +366,173 @@ def use_chi(self, use_chi):
+            Temperature (in units of Kelvin) of the provided dataset.
+
+        """
+        check_type('temperature', temperature, Real)
+
+        temp_store = self.temperatures.tolist().append(temperature)
+        self.temperatures = temp_store
+
+        self._total.append(None)
+        self._absorption.append(None)
+        self._scatter_matrix.append(None)
+        self._multiplicity_matrix.append(None)
+        self._fission.append(None)
+        self._nu_fission.append(None)
+        self._kappa_fission.append(None)
+        self._chi.append(None)
+
+    def set_total(self, total, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        total: nd.nparray
+            Total Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_total_mgxs()
+
+        """
         check_type('total', total, Iterable, expected_iter_type=Real)
         # Convert to a numpy array so we can easily get the shape for
         # checking
         nptotal = np.asarray(total)
         check_value('total shape', nptotal.shape, [self.vector_shape])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = self.temperatures.tolist().index(temperature)
+        self._total[i] = nptotal
+
+    def set_absorption(self, absorption, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
 
-        self._total = nptotal
+        Parameters
+        ----------
+        absorption: nd.nparray
+            Absorption Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_absorption_mgxs()
 
-    @absorption.setter
-    def absorption(self, absorption):
+        """
         check_type('absorption', absorption, Iterable, expected_iter_type=Real)
         # Convert to a numpy array so we can easily get the shape for
         # checking
         npabsorption = np.asarray(absorption)
         check_value('absorption shape', npabsorption.shape,
                     [self.vector_shape])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = self.temperatures.tolist().index(temperature)
+        self._absorption[i] = npabsorption
+
+    def set_fission(self, fission, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
 
-        self._absorption = npabsorption
+        Parameters
+        ----------
+        fission: nd.nparray
```

Comment:
hahahaha sorry about that (i dont know why i find what I did funny, but I do)

- [x]